plugins/Actions/ArchivingHelper.php

```diff
@@ -228,7 +228,8 @@ public static function getRankingQueryLimit()
         $limit = $configLimit == 0 ? 0 : max(
             $configLimit,
             $configGeneral['datatable_archiving_maximum_rows_actions'],
-            $configGeneral['datatable_archiving_maximum_rows_subtable_actions']
+            $configGeneral['datatable_archiving_maximum_rows_subtable_actions'],
+            $configGeneral['datatable_archiving_maximum_rows_site_search']
```

Comment:
@katebutler I wonder if we could leave the general `rankingQueryLimit` untouched and instead only for 

`$this->archiveDayActions($rankingQueryLimit);` apply the possibly higher limit as in `$this->archiveDayActions(max($rankingQueryLimit, $config['datatable_archiving_maximum_rows_site_search'] );` in  https://github.com/matomo-org/matomo/blob/3.9.0-b1/plugins/Actions/Archiver.php#L60 ?

This way we would fetch only the needed rows for all other reports except for the one that generates site search. Eg all other "entry actions, exit actions, actions time, events, ... " would still use the correct limit directly and not fetch more than possibly needed.